### ISSUE TITLE
Weird email sender

### ISSUE THREAD

**Issue by akmet** [tomav/docker-mailserver#853]
Hey! I installed the latest docker-mailserver and nearly everything works fine.
I am able to send and receive mails. but when sending emails the FROM value looks a little bit weird..
My Dockerhost's hostname is id.vendor.com
My Docker's is mail.host.com
I have mydomain.com which MX records points to id.vendor.com
My account sender@mydomain.com exists, sender@host.com not.

Is this an expected behavior? Some clients(rainloop eg) are not displaying the senders name, they display the weird from value. Thunderbird's source code view states from as my original email and return-path as the weird one.

When you need further information please feel free to ask :)

docker_compose.yml:
```
version: '2'

services:
  mail:
    image: tvial/docker-mailserver:latest
    hostname: mail
    domainname: host.com
    container_name: mail
    ports:
    - "25:25"
    - "143:143"
    - "587:587"
    - "993:993"
    - "4190:4190"
    volumes:
    - maildata:/var/mail
    - mailstate:/var/mail-state
    - ./config/:/tmp/docker-mailserver/
    - /etc/letsencrypt/:/etc/letsencrypt/
    - "/etc/timezone:/etc/timezone:ro"
    - "/etc/localtime:/etc/localtime:ro"
    environment:
    - ENABLE_SPAMASSASSIN=1
    - ENABLE_CLAMAV=1
    - ENABLE_FAIL2BAN=1
    - ENABLE_MANAGESIEVE=1
    - ENABLE_POSTGREY=1
    - SSL_TYPE=letsencrypt
    - ONE_DIR=1
    - DMS_DEBUG=0
    - OVERRIDE_HOSTNAME=id.vendor.com
    cap_add:
    - NET_ADMIN
    - SYS_PTRACE
    restart: always
```

Mail log: 
```
Feb 19 14:17:14 mail postfix/submission/smtpd[1715]: 3474D201015: client=client.domain.com[CLIENT_IP], sasl_method=PLAIN, sasl_username=sender@mydomain.com
Feb 19 14:17:14 mail postfix/sender-cleanup/cleanup[1721]: 3474D201015: message-id=<522c6956-3313-e773-06c1-6efdd7e654ee@mydomain.com>
Feb 19 14:17:14 mail postfix/sender-cleanup/cleanup[1721]: 3474D201015: replace: header MIME-Version: 1.0 from client.domain.com[CLIENT_IP]; from=<srs0=tvi6=fn=mydomain.com=sender@host.com> to=<target@target.com> proto=ESMTP helo=<[SOME_IP]>: Mime-Version: 1.0
Feb 19 14:17:14 mail opendkim[257]: 3474D201015: DKIM-Signature field added (s=mail, d=mydomain.com)
Feb 19 14:17:14 mail postfix/qmgr[1133]: 3474D201015: from=<srs0=tvi6=fn=mydomain.com=sender@host.com>, size=357, nrcpt=1 (queue active)
Feb 19 14:17:14 mail postfix/submission/smtpd[1715]: disconnect from client.domain.com[CLIENT_IP] ehlo=2 starttls=1 auth=1 mail=1 rcpt=1 data=1 quit=1 commands=8
Feb 19 14:17:14 mail dovecot: imap(sender@mydomain.com): Logged out in=1574 out=2668
Feb 19 14:17:15 mail postfix/smtpd[1731]: connect from localhost[127.0.0.1]
Feb 19 14:17:15 mail postfix/smtpd[1731]: 16F11201017: client=localhost[127.0.0.1]
Feb 19 14:17:15 mail postfix/cleanup[1732]: 16F11201017: message-id=<522c6956-3313-e773-06c1-6efdd7e654ee@mydomain.com>info
Feb 19 14:17:15 mail postfix/qmgr[1133]: 16F11201017: from=<srs0=tvi6=fn=mydomain.com=sender@host.com>, size=1076, nrcpt=1 (queue active)
Feb 19 14:17:15 mail amavis[1037]: (01037-01) Passed CLEAN {RelayedOpenRelay}, [CLIENT_IP]:24887 [CLIENT_IP] <srs0=tvi6=fn=mydomain.com=sender@host.com> -> <target@target.com>, Queue-ID: 3474D201015, Message-ID: <522c6956-3313-e773-06c1-6efdd7e654ee@mydomain.com>, mail_id: OZYKbL-x9f1B, Hits: 5.356, size: 850, queued_as: 16F11201017, 773 ms
Feb 19 14:17:15 mail postfix/smtp[1724]: 3474D201015: to=<target@target.com>, relay=127.0.0.1[127.0.0.1]:10024, delay=0.93, delays=0.15/0.01/0.01/0.77, dsn=2.0.0, status=sent (250 2.0.0 from MTA(smtp:[127.0.0.1]:10025): 250 2.0.0 Ok: queued as 16F11201017)
Feb 19 14:17:15 mail postfix/qmgr[1133]: 3474D201015: removed
Feb 19 14:17:16 mail postfix/smtp[1735]: Trusted TLS connection established to mail.target.com[TARGET_IP]:25: TLSv1.2 with cipher DHE-RSA-AES256-GCM-SHA384 (256/256 bits)
Feb 19 14:17:16 mail postfix/smtp[1735]: 16F11201017: to=<target@target.com>, relay=mail.target.com[TARGET_IP]:25, delay=1.1, delays=0.02/0.02/0.9/0.14, dsn=2.0.0, status=sent (250 ok:  Message 150692822 accepted)
Feb 19 14:17:16 mail postfix/qmgr[1133]: 16F11201017: removed
```

**Comment by 17Halbe** [comment]
So, to get you right, your email is being sent as sender@host.com and not as it should be as  sender@mydomain.com? But your OVERRIDE_HOSTNAME=id.vendor.com?

**Comment by akmet** [comment]
No. It looks more weird. From looks like this: srs0=tvi6=fn=mydomain.com=sender@host.com

I tested a little bit more and it seems to depend on the account I am using:
When using an valid server@host.com everthing works fine.
When using valid sender@mydomain.com the Return-Path looks crazy, but From seems to be right.
When using valid root@id.vendor.com on my host with msmtp Return-Path and From looks like above.
With valid I mean I created accounts with the setup.sh tool.

It seems that when the sender is authenticated and not ends on my domainname it adds it?
